frontend/coprs_frontend/coprs/auth.py

```diff
@@ -238,8 +238,15 @@ def user_from_username(username, load_metadata=False):
     def _krb5_login_redirect(next_url=None):
         if app.config['KRB5_LOGIN']:
             # Pick the first one for now.
-            return flask.redirect(flask.url_for("apiv3_ns.gssapi_login",
-                                                next=next_url))
+            return flask.redirect(
+                # TODO: this is cursed...
+                #  url_for from flask works with resource class name
+                #  as snake_case. This is bullshit but the flask-restx
+                #  url_for takes as parameter the resource -> cyclic import
+                #  ... the gssapi endpoint depends on the UserAuth
+                #  but how to fix it? :clueless:
```

Comment:
Cyclic imports are stressful.  Let's have a quick chat about how to fix this, instead of this TODO which will never disappear :)

Reply:
Is this a TODO item actually? Referencing routes using a string like this ` flask.url_for("apiv3_ns._gss_api_login")` is legit, so I don't see a problem here.

The only weird thing is IMHO the leading underscore

Reply:
> The only weird thing is IMHO the leading underscore

This can be fixed by e.g.

```diff
diff --git a/frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_general.py b/frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_general.py
index 48feb66b0..79977656f 100644
--- a/frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_general.py
+++ b/frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_general.py
@@ -58,7 +58,7 @@ def krb_straighten_username(krb_remote_user):
     return username


-apiv3_general_ns = Namespace("", description="APIv3 general endpoints")
+apiv3_general_ns = Namespace("general", description="APIv3 general endpoints")
 api.add_namespace(apiv3_general_ns)
```

Then the route name would be `apiv3_ns.general_gss_api_login`. But we don't have to do this. The leading underscore is fine.

Reply:
Good catch! Nice, yes - please - rather be explicit with the Namespace naming.